### ISSUE TITLE
How can I input two different types of graghs into the network at the same time?

### ISSUE THREAD

**Issue by cowry5** [rusty1s/pytorch_geometric#781]
## ❓ Questions & Help

Hi~ I now have a  network input that is two types of graphs，GraphA and GraphB. These two Graphs correspond to the same label. 
So, How do I input these two types of graphs at the same time? 
I have two ideas：
1. Put both Graphs in the same Dataloader, But I don't know how to implement it, it may destroy the graph's structure inside.
2. Use two Dataloaders for GraphA and GraphB，and do this:
```python
for batch1, batch2 in zip(data_loaderA, data_loaderB):
    batch1, batch2
```
But, if I set the parameter of dataloader, `shuffle=True`. These two dataloaders will not generate the same random number, which will cause `batch1, batch2` to lose the corresponding relationship, without the same label.

I really need your advice, thank you!

**Comment by rusty1s** [comment]
I believe the easiest way is to define a new `PairDataset`:
```python
class PairDataset(torch.utils.data.Dataset):
    def __init__(self, datasetA, datasetB):
        ...

    def __getitem__(self, idx):
        return self.datasetA[idx], self.datasetB[idx]
```
and use it in a PyTorch DataLoader as follows:
```python
from torch_geometric.data import Batch

def collate(self, data_list):
    batchA = Batch.from_data_list([data[0] for data in data_list])
    batchB = Batch.from_data_list([data[1] for data in data_list])
    return batchA, batchB

torch.utils.DataLoader(pair_dataset, batch_size=..., shuffle=True, collate_fn=collate)
```

**Comment by cowry5** [comment]
Thank you very much, this solved my problem.

**Comment by YunYunY** [comment]
> I believe the easiest way is to define a new `PairDataset`:
> 
> ```python
> class PairDataset(torch.utils.data.Dataset):
>     def __init__(self, datasetA, datasetB):
>         ...
> 
>     def __getitem__(self, idx):
>         return self.datasetA[idx], self.datasetB[idx]
> ```
> 
> and use it in a PyTorch `DataLoader` as follows:
> 
> ```python
> from torch_geometric.data import Batch
> 
> def collate(self, data_list):
>     batchA = Batch.from_data_list([data[0] for data in data_list])
>     batchB = Batch.from_data_list([data[1] for data in data_list])
>     return batchA, batchB
> 
> torch.utils.DataLoader(pair_dataset, batch_size=..., shuffle=True, collate_fn=collate)
> ```

Hi,
What if there are multiple graphs with different number of node rather than two paired graphs?
Thank you.

**Comment by rusty1s** [comment]
Can you be a bit more specific? Do you want to store multiple graphs in a single `data` object? Or do you simply want to store graphs with varying number of nodes? The latter should be automatically supported.

**Comment by YunYunY** [comment]
Thank you so much. The issue has been solved in [discussion](https://github.com/rusty1s/pytorch_geometric/discussions/2948).
